docs/azureai/azureai-openai-integration.md

```diff
@@ -1,138 +1,384 @@
+### Add an Azure OpenAI client
+
+In the _Program.cs_ file of your client-consuming project, use the <xref:Microsoft.Extensions.Hosting.AspireAzureOpenAIExtensions.AddAzureOpenAIClient(Microsoft.Extensions.Hosting.IHostApplicationBuilder,System.String,System.Action{Aspire.Azure.AI.OpenAI.AzureOpenAISettings},System.Action{Azure.Core.Extensions.IAzureClientBuilder{Azure.AI.OpenAI.AzureOpenAIClient,Azure.AI.OpenAI.AzureOpenAIClientOptions}})> method on any <xref:Microsoft.Extensions.Hosting.IHostApplicationBuilder> to register an `OpenAIClient` for dependency injection (DI). The `AzureOpenAIClient` is a subclass of `OpenAIClient`, allowing you to request either type from DI. This ensures code not dependent on Azure-specific features remains generic. The `AddAzureOpenAIClient` method requires a connection name parameter.
 
 ```csharp
-var builder = DistributedApplication.CreateBuilder(args);
+builder.AddAzureOpenAIClient(connectionName: "openai");
+```
 
-var openai = builder.ExecutionContext.IsPublishMode
-    ? builder.AddAzureOpenAI("openAiConnectionName")
-    : builder.AddConnectionString("openAiConnectionName");
+> [!TIP]
+> The `connectionName` parameter must match the name used when adding the Azure OpenAI resource in the app host project. For more information, see [Add an Azure OpenAI resource](#add-an-azure-openai-resource).
 
-builder.AddProject<Projects.ExampleProject>()
-       .WithReference(openai);
+After adding the `OpenAIClient`, you can retrieve the client instance using dependency injection:
+
+```csharp
+public class ExampleService(OpenAIClient client)
+{
+    // Use client...
+}
 ```
 
-The `AddAzureAIOpenAI` method will read connection information from the app host's configuration (for example, from "user secrets") under the `ConnectionStrings:openAiConnectionName` config key. The <xref:Aspire.Hosting.ResourceBuilderExtensions.WithReference%2A> method passes that connection information into a connection string named `openAiConnectionName` in the `ExampleProject` project. In the _:::no-loc text="Program.cs":::_ file of ExampleProject, the connection can be consumed using:
+For more information, see:
+
+- [Azure.AI.OpenAI documentation](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/openai/Azure.AI.OpenAI/README.md) for examples on using the `OpenAIClient`.
+- [Dependency injection in .NET](/dotnet/core/extensions/dependency-injection) for details on dependency injection.
+- [Quickstart: Get started using GPT-35-Turbo and GPT-4 with Azure OpenAI Service](/azure/ai-services/openai/chatgpt-quickstart?pivots=programming-language-csharp).
+
+### Add Azure OpenAI client with registered `IChatClient`
+
+If you're interested in using the new <xref:Microsoft.Extensions.AI.IChatClient> interface, with the OpenAI client, simply chain either of the following APIs to the `AddAzureOpenAIClient` method:
```

Comment:
I don't like saying things like "The **new** blah interface." Eventually this won't be "new" which means you have to edit your doc to remove this type of wording.